event_registration_mass_confirm/wizard/confirm_attendees.py

```diff
@@ -1,21 +1,20 @@
-# -*- coding: utf-8 -*-
 # Copyright 2018 Camptocamp SA
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
 from odoo import models, api
 
 
-class ConfirmEventRegistrations(models.TransientModel):
+class EventRegistrationMassConfirm(models.TransientModel):
 
-    _name = "event.registration.massvalidation"
+    _name = "event.registration.massconfirm"
 
     @api.multi
-    def validate_registrations(self):
+    def confirm_registrations(self):
         """Validate selected registrations."""
         context = dict(self.env.context or {})
         active_ids = context.get('active_ids', []) or []
 
         registrations = self.env['event.registration'].browse(active_ids)
 
-        registrations.confirm_registration()
-
-        return {'type': 'ir.actions.act_window_close'}
+        if len(registrations) > 0:
+            registrations.confirm_registration()
+        return registrations
```

Comment:
Here you would need to return a server action and not a recordset. You can read an existing one and return it as a dict with registrations.ids in domain.